packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -33,13 +41,20 @@ export const useConditionalFeature = <T extends JSONValue>({
 
   const { data: featureValue, isLoading } = useQuery(
     queryKey,
-    () => {
+    async () => {
       if (!shouldEvaluate) {
         return feature.defaultValue as WidenPrimitives<T>;
       }
-      return growthbook
+
+      const value = await (growthbook
         ? growthbook.getFeatureValue(feature.id, feature.defaultValue)
-        : (feature.defaultValue as WidenPrimitives<T>);
+        : Promise.resolve(feature.defaultValue as WidenPrimitives<T>));
+
+      if (onAfterEvaluation) {
+        onAfterEvaluation(value);
+      }
```

Comment:
Introduced this so we won't have to introduce a `useEffect` whenever we need to do something after the evaluation. For example, the modal would have to be called in a `useEffect` if we didn't have this.

Reply:
Example:
- hook A mounts to check the value
- query resolves but onAfterEvaluation is not called since hook A does not need it
- hook B mounts to check the value with onAfterEvaluation
- query was already resolved so no onAfterEvaluation is called
- hook B logic never executes

Reply:
I think it would still execute since hook B won't have the correct value unless it goes through that code. Are we probably missing an early return of the value (client.getData) if it has been fetched already? Which would then make the scenario you mentioned, valid?

Reply:
> I think it would still execute since hook B won't have the correct value unless it goes through that code

I think not, `useQuery` will just return cached value per query key which would be the same for both hooks.

Reply:
> Are we probably missing an early return of the value (client.getData) if it has been fetched already? Which would then make the scenario you mentioned, valid?

Again, if this happens inside `queryFn` it would still not execute because useQuery would just use cached value without calling a function.

Reply:
With your comment of useEffect, when would you theoretically want `onAfterEvaluation` to execute? Any time the feature value changes or only once per hook call?

Reply:
> I think not, useQuery will just return cached value per query key which would be the same for both hooks.

Ah, yes. That makes sense.

> With your comment of useEffect, when would you theoretically want onAfterEvaluation to execute? Any time the feature value changes or only once per hook call?

In this case, it is probably better to do a useEffect for now since it is more predictable to say whenever the value pulled is correct, we should run the function. Open to suggestions, but for now, I will have to resort to `useEffect`, since I believe the `onSuccess` for `useQuery` will soon be obsolete.

Reply:
Think so yes, only alternative that we can look is that if we can toggle modal from `onCardBookmark`?

Reply:
We can, so instead of calling `setJustBookmarked(true);` in `onCardBookmark` we call `growthbook.getFeatureValue` there and then we can toggle modal depending on returned value then and there. 

So in `onCardBookmark`:
```js
if (!post.bookmarked && !seenBookmarkPromotion && !justBookmarked) {
  const featureValue = growthbook.getFeatureValue(feature.bookmarkLoops)

  if (featureValue) {
    completeAction(ActionType.BookmarkPromoteMobile);
    openModal({
      type: LazyModal.MarketingCta,
      props: { marketingCta: promotion.bookmarkPromoteMobile },
    });
  }
}
```

Reply:
Yeah, that's true, we can do this. My only concern is the usage of `growthbook.getFeatureValue` is not often how we enroll to features. It might have some confusion on which to use and when for others in the future, but yeah, for me, that works alright. If everyone agrees, then I'll be cool with it.

Reply:
I think either you should use Antes component here or do the eval top level.
This hook should be as lean as possible (just my 2 cents without context )

Ante should your component not do with 2 buttons one with the exp one not?

Reply:
We can do it in multiple ways yes, but I think enrolment wise it would be most accurate to enrol user when he actually bookmarks and not anytime he sees bookmark button?

Reply:
> We can do it in multiple ways yes, but I think enrolment wise it would be most accurate to enrol user when he actually bookmarks and not anytime he sees bookmark button?

Yes, that's why we are trying to only decide to enroll when bookmark is executed. In that case, I will change the implementation to use `growthbook.getFeatureValue`.

Reply:
Hmm yeah, should then be as simple as direct eval.
Although maybe youll need another wrapper to avoid too many deps

Reply:
Deps are looking good.

Should now be updated as to what was discussed :shipit: